packages/@glimmer/program/lib/program.ts

```diff
@@ -71,14 +74,26 @@ export class Heap implements CompileTimeHeap {
       this.table = table;
       this.offset = this.heap.length;
       this.handle = handle;
+      this.pageSize = this.heap.length;
```

Comment:
As `this.pageSize` keeps track of the used space in the last page of the heap, I think this will only work if `serializedHeap.buffer` is a multiple of `PAGE_SIZE`.
If in stead, we keep track of the free space, then this should work.